shimmingtoolbox/shim/sequencer.py

```diff
@@ -833,26 +834,33 @@ def select_optimizer(method, unshimmed, affine, coils: ListCoil, pmu: PmuResp =
     return optimizer
 
 
-_optimize_scope = {}  # i?: (_optimizer: Optimizer, nii_mask_anat, slices_anat, shimwise_bounds=None,
-                      #     dilation_kernel='sphere', dilation_size=3, path_output=None) 
-def _optimize(_optimizer: Optimizer, nii_mask_anat, slices_anat, shimwise_bounds=None,
-              dilation_kernel='sphere', dilation_size=3, path_output=None):
+# Used as a global dictionary to pass argument to the _opt function. This avoids to pickle the data resulting in a slow
+# start of the multiprocessing pool
+# uuid: (_optimizer: Optimizer, nii_mask_anat, slices_anat, shimwise_bounds=None, dilation_kernel='sphere',
+# dilation_size=3, path_output=None)
+_optimize_scope = {}
 
+
+def _optimize(optimizer: Optimizer, nii_mask_anat, slices_anat, shimwise_bounds=None,
+              dilation_kernel='sphere', dilation_size=3, path_output=None):
     # Count shims to perform
     n_shims = len(slices_anat)
-    
-    global _optimize_scope
-    global optimizer
-    optimizer = _optimizer # not a super safe thing to do
 
     # multiprocessing optimization
     mp.set_start_method('fork', force=True)
     run_id = uuid.uuid4().hex
-    _optimize_scope[run_id] = (optimizer, nii_mask_anat, slices_anat, dilation_kernel, dilation_size, path_output, shimwise_bounds)
-    with mp.Pool(mp.cpu_count()) as pool:
-        del _optimize_scope[run_id] # should be safe to del here? because at this point all the child processes have forked and inherited their copy
-        results = pool.starmap_async(_opt, [(run_id, i) for i in range(n_shims)]).get()
-    
+    global _optimize_scope
+    _optimize_scope[run_id] = (
+        optimizer, nii_mask_anat, slices_anat, dilation_kernel, dilation_size, path_output, shimwise_bounds)
+    try:
+        with mp.Pool(mp.cpu_count()) as pool:
```

Comment:
`cpu_count()` is the default:

```suggestion
        with mp.Pool() as pool:
```

Reply:
Resolved here: [2a09aa8](https://github.com/shimming-toolbox/shimming-toolbox/pull/390/commits/2a09aa8f1a44ce36378bbfef174bc3ecca073140)